### PR TITLE
Give extra architecture information

### PR DESCRIPTION
**Pull request by SimonWiet**
Adds the architecture diagram and some more detailed description on how to set up a world

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,6 +63,18 @@ A `world` is the definition of the environment for a specific application. To se
      - `getHighlightAtPosition(int x, int y)` is called after a click on the coordinates in the parameters. If something implementing `IHighlightable` is clicked, this should be returned to display its information. If `null` is returned, the current selection is kept.
      - `close()` is called when the simulator is closed. Clean up everything that needs to be cleaned up.
 
+   - When there is the need to add/edit static parameters (magic numbers), do so by extending the class `core.Configuration`
```
Comment (christianzoellner) [inline README.md:66]:
```suggestion
   - When you need to overwrite and/or add static parameters (magic numbers), do so by adding a class that extends `core.Configuration` in your `world` subpackage.

```

**Review by christianschaeffer — CHANGES_REQUESTED** [review]
Generally this looks good. I proposed some typo fixes however. 

Also, please make sure you write "statechart" consistently as one word (as in the paper). I read a lot of "state chart". Thanks!

**Comment by SimonWiet** [comment]
Valid comments, I commited them and additionally renamed some occurences of "state chart(s)" to "statechart(s)"

**Review by christianschaeffer — APPROVED** [review]
Thanks for resolving my previous comments and for your work on the text @SimonWiet!

**Pull request merged**